instrumentation/rocketmq-client/src/main/java/brave/rocketmq/client/TracingSendMessage.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright The OpenZipkin Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package brave.rocketmq.client;
+
+import brave.Span;
+import brave.Tracer;
+import brave.internal.Nullable;
+import brave.messaging.MessagingRequest;
+import brave.propagation.CurrentTraceContext;
+import brave.propagation.TraceContext;
+import brave.propagation.TraceContextOrSamplingFlags;
+import brave.sampler.SamplerFunction;
+import io.opentelemetry.api.internal.StringUtils;
+import org.apache.rocketmq.client.hook.SendMessageContext;
+import org.apache.rocketmq.client.hook.SendMessageHook;
+import org.apache.rocketmq.client.impl.CommunicationMode;
+import org.apache.rocketmq.client.producer.SendResult;
+import org.apache.rocketmq.common.message.Message;
+
+import static brave.Span.Kind.PRODUCER;
+import static brave.rocketmq.client.RocketMQTracing.ROCKETMQ_TAGS;
+import static brave.rocketmq.client.RocketMQTracing.ROCKETMQ_TOPIC;
+
+/**
+ * For the user side, there are many overloaded methods for send message in
+ * {@link org.apache.rocketmq.client.producer.MQProducer}, so implementing
+ * {@link org.apache.rocketmq.client.hook.SendMessageHook} might be an efficient approach to enable tracing.
+ */
+final class TracingSendMessage implements SendMessageHook {
```

Comment:
more simple naming convention is to just put Tracing in front

```suggestion
final class TracingSendMessageHook implements SendMessageHook {
```